### DISCUSSION TITLE
AR tag detection

### DISCUSSION THREAD

**Discussion by Aanal2901** [larics/icuas22_competition#25] [Q&A]
@RobertMilijas 
1) To flag the detection of marker, we are supposed to publish an annotated image. What exactly are we supposed to annotate, the center or bounding box of AR tag? Are we supposed to publish one image or update in real time?
2) For evaluating the accuracy of ball with respect to target, is the distance being calculated with respect to the center of AR tag?
3) Do we publish the reconstructed position from the real position of the target on a pre-defined topic?

**Comment by fpetric** [comment]
hi @Aanal2901, 

me again, just to start the discussion, @RobertMilijas may provide additional insight:) 

1. You should publish an image of the ARTag from the camera, with the bounding box as a proof that your detection algorithm works. This image should be published only once. 
2. I am not sure where the origin of the tag is, but everything will be calculated with respect to the origin of the tag (whether it is the center or one of the corners, I am not sure at the moment. @RobertMilijas please help)
3. Once you detect the tag, you will need to calculate where it is in the frame of the camera (you may use off-the-shelf algorithm or develop your own for increased precision), and then transform the pose of the tag from the camera frame to the gazebo world frame. You will not know the "real position" of the target, but you will publish your estimate of the tag pose on a given topic (the topics to be used for image and pose will be announced once we finish developing the evaluation setup)

**Reply by RobertMilijas** [reply]
Hi @Aanal2901 and @fpetric,

sorry for the long wait. I can confirm that the origin of the tag is in the center of the image. I've marked the right-hand coordinate frame in this figure:

![alvar_coordinate_frame](https://user-images.githubusercontent.com/24401245/158265646-612a5edf-f83f-4eca-80d6-1293e9fb9715.png)
